src/hw/fourth/MyArrayList.java

```diff
@@ -0,0 +1,126 @@
+package hw.fourth;
+
+public class MyArrayList<T> implements List<T> {
+
+    private int size;
+    private T[] array;
+
+    public MyArrayList() {
+        size = 0;
+        array = (T[]) new Object[size];
+    }
+
+    public MyArrayList(T[] array) {
+        this.array = array;
+        size = array.length;
+    }
+
+    @Override
+    public void add(T value) {
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+        } else {
+            T[] newArray = (T[]) new Object[size + 5];
+            for (int i = 0; i < size; i++) {
+                newArray[i] = array[i];
+            }
+            newArray[size] = value;
+            array = newArray;
+            size++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index == (size - 1)) {
+            add(value);
+        } else {
+            int sizeNewArray;
+            if (size < array.length) {
+                sizeNewArray = array.length;
+            } else {
+                sizeNewArray = size + 5;
+            }
+            T[] newArray = (T[]) new Object[sizeNewArray];
+            int element = 0;
+            for (int i = 0; i < size + 1; i++) {
+                if (i == index) {
+                    newArray[i] = value;
+                } else {
+                    newArray[i] = array[element];
+                    element++;
+                }
+            }
+            array = newArray;
+            size++;
+        }
+    }
+
+    @Override
+    public int size() {
+        return this.size;
+    }
+
+    @Override
+    public T get(int index) {
+        return array[index];
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        if (list.size() == 1){
```

Comment:
you should write something general, without check if list.size() == 1

Reply:
That much general:
`public void addAll(List<T> list) {
        for (int i = 0; i < list.size(); i++) {
            this.add(list.get(i));
        }
 }`
Or it's better to not use already written methods?